### DISCUSSION TITLE
[OP Stack Deployment] Op-Geth giving continuous warnings

### DISCUSSION THREAD

**Discussion by priyanshuthapliyal55** [ethereum-optimism/developers#433] [Deploying the OP Stack]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

getting these warning in my replica node (Non-sequencer node) although able fetch block from this replica node. And also this warning start coming after my sequencer node suddenly stop and was not able to connect to my holesky node so i have restart my Sequencer node then restarted my replica node but replica node is constantly giving this warning message and also i deployed blockexplorer connected with replica node and can see the correct latest Block. I am deploying testnet rollups with custom gas token.

### Logs

![image](https://github.com/ethereum-optimism/developers/assets/159440241/20f47e80-84ee-4ce5-9a6a-c38b4df300d0)

**Comment by hash1go** [comment]
Hello.
Could you please provide the configuration and version information?

**Reply by priyanshuthapliyal55** [reply]
My sequencer node is running fine even this replica node is working fine because it is showing blocks being generated in explorer but these logs are taking too much space and also i want to see proper logs instead of these warning.

For Op-geth i am using tag op-geth v1.101315.1
and using this conf for starting Op-geth
./op-geth/build/bin/geth \
  --datadir /home/op-geth/datadir \
  --http \
  --http.corsdomain="*" \
  --http.vhosts="*" \
  --http.addr=0.0.0.0 \
  --http.api=web3,debug,eth,txpool,net,engine \
  --ws \
  --ws.addr=0.0.0.0 \
  --ws.port=8546 \
  --ws.origins="*" \
  --ws.api=debug,eth,txpool,net,engine \
  --syncmode=full \
  --gcmode=full \
  --nodiscover \
  --maxpeers=0 \
  --networkid=42069 \
  --authrpc.vhosts="*" \
  --authrpc.addr=0.0.0.0 \
  --authrpc.port=8551 \
  --authrpc.jwtsecret=/home/op-geth/jwt.txt \
 ----rollup.sequencerhttp=sequencer rpc
  --rollup.disabletxpoolgossip=true
  
For opnode using op-node version v1.7.7-rc.2-926ba712-1716907690

home/zeever/optimism/op-node/bin/op-node \
  --l2=http://localhost:8551 \
  --l2.jwt-secret=/home/optimism/op-node/jwt.txt \
  --sequencer.enabled \
  --sequencer.l1-confs=5 \
  --verifier.l1-confs=4 \
  --rollup.config=/home/optimism/op-node/rollup.json \
  --rpc.addr=0.0.0.0 \
 --p2p2.static= 
  --p2p.listen.ip=0.0.0.0 \
  --p2p.listen.tcp=9003 \
  --p2p.listen.udp=9003 \
  --rpc.port=8547 \
  --metrics.addr=0.0.0.0 \
  --metrics.enabled=true \
  --metrics.port=7300 \
  --rpc.enable-admin \
  --p2p.sequencer.key=privatekey \
  --l1.beacon=holesky beacon endpoint \
  --l1=holesky rpc \
  --l1.rpckind=any \
  --log.level=info

**Reply by Chomtana** [reply]
It feels like you are using a full node but receiving `eth_getBalance` for an old block that only exists on an archive node.

> Deployed block explorer connected with replica node.

I remember that Blockscout requires an archive node, so you need to set `--gcmode=archive`.

**Reply by priyanshuthapliyal55** [reply]
> It feels like you are using a full node but receiving `eth_getBalance` for an old block that only exists on an archive node.
> 
> > Deployed block explorer connected with replica node.
> 
> I remember that Blockscout requires an archive node, so you need to set `--gcmode=archive`.

oh i didnt know that will try to do this so can i just change --gcmode=archive in configuration directly or do i need to initilize geth again like settingup op-geth again?

**Reply by Chomtana** [reply]
You need to init op-geth again to change from full to archive node and vice versa.

Blockscout can operate on full node if you only want them to index latest blocks but it will keep trying to call archive node only feature that should cause these warnings.

**Reply by priyanshuthapliyal55** [reply]
> You need to init op-geth again to change from full to archive node and vice versa.
> 
> Blockscout can operate on full node if you only want them to index latest blocks but it will keep trying to call archive node only feature that should cause these warnings.

Thanks it worked but now the issue is i am not able to do transaction using this rpc endpoint nor with sequencer rpc endpoint? What could be the issue? L2 is working with same conf i have mentioned above but cant do transaction from l2 to l2

**Reply by opfocus** [reply]
Enabling the sequencer on the op-node of the replica node may cause problems

```
./bin/op-node \
  --l1=$L1_RPC_URL \
  --l1.rpckind=$L1_RPC_KIND \
  --l2=ws://localhost:8551 \
  --l2.jwt-secret=./jwt.txt \
  --rollup.config=./rollup.json \
  --p2p.static=<sequencer node multiaddr> \
  --p2p.listen.ip=0.0.0.0 \
  --p2p.listen.tcp=9003 \
  --p2p.listen.udp=9003
```

**Reply by priyanshuthapliyal55** [reply]
> Enabling the sequencer on the op-node of the replica node may cause problems
> 
> ```
> ./bin/op-node \
>   --l1=$L1_RPC_URL \
>   --l1.rpckind=$L1_RPC_KIND \
>   --l2=ws://localhost:8551 \
>   --l2.jwt-secret=./jwt.txt \
>   --rollup.config=./rollup.json \
>   --p2p.static=<sequencer node multiaddr> \
>   --p2p.listen.ip=0.0.0.0 \
>   --p2p.listen.tcp=9003 \
>   --p2p.listen.udp=9003
> ```

But I did not enable it. It's a non sequencer node. Above I might have given sequencer enabled by mistake but it is not in my replica node.

**Reply by Chomtana** [reply]
You have set the wrong sequencer endpoint? It's the RPC endpoint of the primary op-geth, not op-node.

```
--rollup.sequencerhttp="<Primary RPC endpoint>"
```

**Reply by priyanshuthapliyal55** [reply]
> You have set the wrong sequencer endpoint? It's the RPC endpoint of the primary op-geth, not op-node.
> 
> ```
> --rollup.sequencerhttp="<Primary RPC endpoint>"
> ```

Yes it's Sequencer Op-geth rpc. I am running L2 chain with Custom gas token. In genesis I have only enable regolith not ecotone or fjord.

**Reply by opfocus** [reply]
> ut I did not enable it. It's a non sequencer node. Above I might have given sequencer enabled by mistake but it is not in my replica node.

Please edit your issue description
